### ISSUE TITLE
Poor sprite quality

### ISSUE THREAD

**Issue by rtalon83** [maplibre/maplibre-gl-js#4242]
I have been researching this topic, and in the forums, it is recommended to process the images so that in SDF mode: true, quality is not lost. However, I am still facing the same problem. Here is how the sprites continue to appear on the map.

Is anyone working on this issue?

**maplibre-gl-js version**: 4.3.2

**browser**: Versión 125.0.6422.142

### Steps to Trigger Behavior

 1. Load a sprite atlas

![image](https://github.com/maplibre/maplibre-gl-js/assets/12005260/1c66101d-5c14-4e5c-ace9-717ebc403c03)

 2. set the icon from atlas

![image](https://github.com/maplibre/maplibre-gl-js/assets/12005260/7923873e-c4be-4533-b9a6-b52bf664f4cc)
 
 3. Can you see the difference between atlas and browser

ATLAS
![image](https://github.com/maplibre/maplibre-gl-js/assets/12005260/2c27cb65-88c1-4b29-ba10-a0b096f77b86)

BROWSER
![image](https://github.com/maplibre/maplibre-gl-js/assets/12005260/84e786dc-f8b6-4813-af91-d54b1d9453d1)

### ATLAS JSON

`"icon": {     
            "x": 2,
            "y": 38,
            "width": 32,
            "height": 32,
            "pixelRatio": 1,
            "sdf": true
        },`

**Comment by HarelM** [comment]
It's hard to give any advice when there's no reproduction and not enough info.
Also, sdf is not a miracle solution and a 32 pixels icon can't be seen too big and keep being sharp...

**Comment by rtalon83** [comment]
Is there any alternative to dynamically change icon-color?

**Comment by HarelM** [comment]
No, but you can load specific images if you want them bigger lazily, or increase the sprint images' size.

**Comment by johncarmack1984** [comment]
Expected SDF behavior when scaling a 32px source up, answered 2024-06-06. This one is resolved and can be closed.
